src/main/java/picard/sam/util/ReadNameParser.java

```diff
@@ -27,6 +29,8 @@
      */
     public static final String DEFAULT_READ_NAME_REGEX = "<optimized capture of last three ':' separated fields as numeric values>".intern();
 
+    private Map<String, PhysicalLocation> locationMap = new HashMap<>();
```

Comment:
hmmm. I think that storing a map with all the names of the reads is not the best design....I suspect that each read will be reads, queried a few different ways (tile, x, y, etc) and then move on. so perhaps use a Cache instead of a full map? see picard/sam/SamErrorMetric/ReadBaseStratification.java:272 for an example.